### ISSUE TITLE
False positive `differing-param-doc` if positional-only

### ISSUE THREAD

**Issue by spagh-eddie** [PyCQA/pylint#6950]
### Bug description

I have a docparam for `bar`, but pylint complains. 
If I make `bar` not positional-only and remove `, /` from function definition, pylint does not complain. 

```python
# pylint: disable=missing-module-docstring,disallowed-name
def foo(bar: int, /):
    """Foo a bar

    :param bar: not spam
    """
    return bar + 1
```

I did a bit of digging myself and saw that `bar` was not included [in this `arguments_node.args`](https://github.com/PyCQA/pylint/blob/f6b2cfeb76dcd571ca02c3853ca80ee3c938e98b/pylint/extensions/docparams.py#L525-L526), but it is included in `arguments_node.arguments`. I am not familiar enough with these to say anything like "it should be `.arguments` instead of `.args`."

### Configuration

```ini
# pyproject.toml
[tool.pylint.master]
load-plugins = [
    "pylint.extensions.docparams",
    "pylint.extensions.docstyle",
]
```

### Command used

```shell
pylint --score no test.py
```

### Pylint output

```shell
************* Module test
test.py:2:0: W9017: "bar" differing in parameter documentation (differing-param-doc)
```

### Expected behavior

expect no output, same as if remove `, /`

### Pylint version

```shell
pylint 2.14.1
astroid 2.11.5
Python 3.10.1 (main, Jan 12 2022, 15:59:59) [Clang 13.0.0 (clang-1300.0.27.3)]
```

### OS / Environment

macOS 12.0.1 (21A559)

**Comment by mpernigo** [comment]
I have done some digging, too.
I think `posonly` could be handled similar to how `kwonly` are ([docparams.py#L527](https://github.com/PyCQA/pylint/blob/main/pylint/extensions/docparams.py#L527), [docparams.py#L542-L544](https://github.com/PyCQA/pylint/blob/main/pylint/extensions/docparams.py#L542-L544)).

I could open a PR for this - unless anybody already started working on it :+1:
